qiskit/algorithms/eigen_solvers/numpy_eigen_solver.py

```diff
@@ -263,3 +322,64 @@ def compute_eigenvalues(
 
         logger.debug("EigensolverResult:\n%s", self._ret)
         return self._ret
+
+    def compute_transition_amplitudes(
+        self,
+        aux_operators: Optional[ListOrDict[OperatorBase]],
```

Comment:
Maybe another naive question: why don't you allow `operator` to be provided here?
Simply because it is not required and can be inserted into `aux_operators`? If so, maybe you should call this argument simply `operators`.

Reply:
For me `operator` refers to the Hamiltonian which is diagonal in the basis of the excited states. Thus I could supply it to `compute_transition_amplitudes` by inserting into `aux_operators` but it would give 0. This transition method really only make sense for certain operators which would almost always be custom user-defined operators.
And for the naming, `aux_operators` matches the interface of _eval_aux_operators or eval_observables so I was trying to keep that as is.